src/Algebra/LabelledGraph.agda

```diff
@@ -23,24 +22,24 @@ infixl 9  _[_]>_
 infix  10 _⊆_
 
 -- Equational theory of graphs
-data _≡_ {A} : (x y : LabelledGraph A) -> Set where
+data _≡_ {A D eq} {d : Dioid D eq} : (x y : LabelledGraph d A) -> Set where
     -- Equivalence relation
-    reflexivity  : ∀ {x     : LabelledGraph A} ->                   x ≡ x
-    symmetry     : ∀ {x y   : LabelledGraph A} -> x ≡ y ->          y ≡ x
-    transitivity : ∀ {x y z : LabelledGraph A} -> x ≡ y -> y ≡ z -> x ≡ z
+    reflexivity  : ∀ {x     : LabelledGraph d A} ->                   x ≡ x
+    symmetry     : ∀ {x y   : LabelledGraph d A} -> x ≡ y ->          y ≡ x
+    transitivity : ∀ {x y z : LabelledGraph d A} -> x ≡ y -> y ≡ z -> x ≡ z
 
     -- Congruence
-    left-congruence  : ∀ {x y z : LabelledGraph A} {r : Dioid} -> x ≡ y -> x [ r ]> z ≡ y [ r ]> z
-    right-congruence : ∀ {x y z : LabelledGraph A} {r : Dioid} -> x ≡ y -> z [ r ]> x ≡ z [ r ]> y
+    left-congruence  : ∀ {x y z : LabelledGraph d A} {r : D} -> x ≡ y -> x [ r ]> z ≡ y [ r ]> z
+    right-congruence : ∀ {x y z : LabelledGraph d A} {r : D} -> x ≡ y -> z [ r ]> x ≡ z [ r ]> y
 
```

Comment:
Are we missing a cong constructor that takes dioid equivalence (eq) into account?

Reply:
Yes, we are, but thanks to `label-addition` it would suffice to change the dioid axiom `+idempotence : ∀ {r : A} -> (r + r) ≡ r` to `+idempotence : ∀ {r s : A} -> r ≡ s -> (r + s) ≡ r`; which should hold for any dioid anyway but isn't currently covered by the laws. What do you think @snowleopard @Rotsor?

Reply:
I guess that works, but it just seems weird. I think the purpose of a "cong" constructor would be much more clear.